### ISSUE TITLE
Screen freezing when several files are opened and firefox is running

### ISSUE THREAD

**Issue by mprt1966** [linuxmint/cinnamon-settings-daemon#295]
```
 * csd version ('dpkg --list | grep cinnamon-settings-daemon' in Mint/Ubuntu)
 * Distribution - Mint 19.3 cinnemon
 * Graphics hardware *and* driver used : nvidia Geforce GTX 550 ti, i use xserver-xorg-video-nouveau cause the recommended driver NVIDIA 440 
[crash.tar.gz](https://github.com/linuxmint/cinnamon-settings-daemon/files/4975740/crash.tar.gz)
does not detect or recognize the dual screen (stay in laptop mode)
 * 32 or 64 bit : 64 bits i7 2600, 16Go DDR3 RAM
 * 
[crash.tar.gz](https://github.com/linuxmint/cinnamon-settings-daemon/files/4975748/crash.tar.gz)
[crash.tar.gz](https://github.com/linuxmint/cinnamon-settings-daemon/files/4975752/crash.tar.gz)

PC : HP pavilion HPE with a SSD drive 250 Go and a second one HDD 2 To

**Issue**
The screen freezes randomly. When firefox (you tube and others) and various files are opened, I move the mouse and the screen freezes. I have to reboot the PC.


**Steps to reproduce**
Maybe open firefox, youtube, open libreoffice writer, access to a second internal drive


**Expected behaviour**
Obviously no freezing screen


**Other information**
===================================================================
 Info                                                              
===================================================================
           PID: 1571 (csd-keyboard)
           UID: 1000 (perrot)
           GID: 1000 (perrot)
        Signal: 5 (TRAP)
     Timestamp: Sat 2020-07-25 16:13:16 CEST (18min ago)
  Command Line: /usr/lib/x86_64-linux-gnu/cinnamon-settings-daemon/csd-keyboard
    Executable: /usr/lib/x86_64-linux-gnu/cinnamon-settings-daemon/csd-keyboard
 Control Group: /user.slice/user-1000.slice/session-c2.scope
          Unit: session-c2.scope
         Slice: user-1000.slice
       Session: c2
     Owner UID: 1000 (perrot)
       Boot ID: 91a3f096046d409785f10afb56ad09ca
    Machine ID: aaa642a1efb049fba8d6ca0c8d22ff0a
      Hostname: perrot-h8-1140fr
       Storage: /var/lib/systemd/coredump/core.csd-keyboard.1000.91a3f096046d409785f10afb56ad09ca.1571.1595686396000000.lz4
       Message: Process 1571 (csd-keyboard) of user 1000 dumped core.
                
                Stack trace of thread 1571:
                #0  0x00007fcee56abea1 n/a (libglib-2.0.so.0)
                #1  0x00007fcee56ae819 g_log_writer_default (libglib-2.0.so.0)
                #2  0x00007fcee56aca8e g_log_structured_array (libglib-2.0.so.0)
                #3  0x00007fcee56ad4ce g_log_structured_standard (libglib-2.0.so.0)
                #4  0x00007fcee6719c41 n/a (libgdk-3.so.0)
                #5  0x00007fcee6726ac3 n/a (libgdk-3.so.0)
                #6  0x00007fcee5dce8fa _XError (libX11.so.6)
                #7  0x00007fcee5dcb82b n/a (libX11.so.6)
                #8  0x00007fcee5dcb8d5 n/a (libX11.so.6)
                #9  0x00007fcee5dcc830 _XReply (libX11.so.6)
                #10 0x00007fcee5db2435 XGetWindowProperty (libX11.so.6)
                #11 0x00007fcee6737064 n/a (libgdk-3.so.0)
                #12 0x00007fcee6737d55 n/a (libgdk-3.so.0)
                #13 0x00007fcee6720eef n/a (libgdk-3.so.0)
                #14 0x00007fcee67211f6 n/a (libgdk-3.so.0)
                #15 0x00007fcee66ebd70 gdk_display_get_event (libgdk-3.so.0)
                #16 0x00007fcee6720f82 n/a (libgdk-3.so.0)
                #17 0x00007fcee56a6417 g_main_context_dispatch (libglib-2.0.so.0)
                #18 0x00007fcee56a6650 n/a (libglib-2.0.so.0)
                #19 0x00007fcee56a6962 g_main_loop_run (libglib-2.0.so.0)
                #20 0x00007fcee6bdea25 gtk_main (libgtk-3.so.0)
                #21 0x00005592e1f9bf19 main (csd-keyboard)
                #22 0x00007fcee506fb97 __libc_start_main (libc.so.6)
                #23 0x00005592e1f9c06a _start (csd-keyboard)
                
                Stack trace of thread 1600:
                #0  0x00007fcee5162cf9 __GI___poll (libc.so.6)
                #1  0x00007fcee56a65c9 n/a (libglib-2.0.so.0)
                #2  0x00007fcee56a6962 g_main_loop_run (libglib-2.0.so.0)
                #3  0x00007fcee63e9276 n/a (libgio-2.0.so.0)
                #4  0x00007fcee56ce175 n/a (libglib-2.0.so.0)
                #5  0x00007fcee2d276db start_thread (libpthread.so.0)
                #6  0x00007fcee516fa3f __clone (libc.so.6)
                
                Stack trace of thread 1599:
                #0  0x00007fcee5162cf9 __GI___poll (libc.so.6)
                #1  0x00007fcee56a65c9 n/a (libglib-2.0.so.0)
                #2  0x00007fcee56a66dc g_main_context_iteration (libglib-2.0.so.0)
                #3  0x00007fcee56a6721 n/a (libglib-2.0.so.0)
                #4  0x00007fcee56ce175 n/a (libglib-2.0.so.0)
                #5  0x00007fcee2d276db start_thread (libpthread.so.0)
                #6  0x00007fcee516fa3f __clone (libc.so.6)
                
                Stack trace of thread 1661:
                #0  0x00007fcee5162cf9 __GI___poll (libc.so.6)
                #1  0x00007fcee56a65c9 n/a (libglib-2.0.so.0)
                #2  0x00007fcee56a66dc g_main_context_iteration (libglib-2.0.so.0)
                #3  0x00007fced909936d n/a (libdconfsettings.so)
                #4  0x00007fcee56ce175 n/a (libglib-2.0.so.0)
                #5  0x00007fcee2d276db start_thread (libpthread.so.0)
                #6  0x00007fcee516fa3f __clone (libc.so.6)
===================================================================
 GDB Log                                                           
===================================================================
[New LWP 1571]
[New LWP 1600]
[New LWP 1599]
[New LWP 1661]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `/usr/lib/x86_64-linux-gnu/cinnamon-settings-daemon/csd-keyboard'.
Program terminated with signal SIGTRAP, Trace/breakpoint trap.
#0  0x00007fcee56abea1 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
[Current thread is 1 (Thread 0x7fcee76aba80 (LWP 1571))]

===================================================================
 GDB Backtrace                                                     
===================================================================
[New LWP 1571]
[New LWP 1600]
[New LWP 1599]
[New LWP 1661]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `/usr/lib/x86_64-linux-gnu/cinnamon-settings-daemon/csd-keyboard'.
Program terminated with signal SIGTRAP, Trace/breakpoint trap.
#0  0x00007fcee56abea1 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
[Current thread is 1 (Thread 0x7fcee76aba80 (LWP 1571))]
#0  0x00007fcee56abea1 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#1  0x00007fcee56ae819 in g_log_writer_default () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007fcee56aca8e in g_log_structured_array () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007fcee56ad4ce in g_log_structured_standard () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#4  0x00007fcee6719c41 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#5  0x00007fcee6726ac3 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#6  0x00007fcee5dce8fa in _XError () from /usr/lib/x86_64-linux-gnu/libX11.so.6
#7  0x00007fcee5dcb82b in ?? () from /usr/lib/x86_64-linux-gnu/libX11.so.6
#8  0x00007fcee5dcb8d5 in ?? () from /usr/lib/x86_64-linux-gnu/libX11.so.6
#9  0x00007fcee5dcc830 in _XReply () from /usr/lib/x86_64-linux-gnu/libX11.so.6
#10 0x00007fcee5db2435 in XGetWindowProperty () from /usr/lib/x86_64-linux-gnu/libX11.so.6
#11 0x00007fcee6737064 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#12 0x00007fcee6737d55 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#13 0x00007fcee6720eef in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#14 0x00007fcee67211f6 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#15 0x00007fcee66ebd70 in gdk_display_get_event () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#16 0x00007fcee6720f82 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#17 0x00007fcee56a6417 in g_main_context_dispatch () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#18 0x00007fcee56a6650 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#19 0x00007fcee56a6962 in g_main_loop_run () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#20 0x00007fcee6bdea25 in gtk_main () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#21 0x00005592e1f9bf19 in main ()

===================================================================
 GDB Backtrace (all threads)                                       
===================================================================
[New LWP 1571]
[New LWP 1600]
[New LWP 1599]
[New LWP 1661]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `/usr/lib/x86_64-linux-gnu/cinnamon-settings-daemon/csd-keyboard'.
Program terminated with signal SIGTRAP, Trace/breakpoint trap.
#0  0x00007fcee56abea1 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
[Current thread is 1 (Thread 0x7fcee76aba80 (LWP 1571))]

Thread 4 (Thread 0x7fced9091700 (LWP 1661)):
#0  0x00007fcee5162cf9 in __GI___poll (fds=0x5592e39f2250, nfds=1, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
        resultvar = 18446744073709551100
        sc_cancel_oldtype = 0
        sc_ret = <optimized out>
#1  0x00007fcee56a65c9 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#2  0x00007fcee56a66dc in g_main_context_iteration () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#3  0x00007fced909936d in ?? () from /usr/lib/x86_64-linux-gnu/gio/modules/libdconfsettings.so
No symbol table info available.
#4  0x00007fcee56ce175 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#5  0x00007fcee2d276db in start_thread (arg=0x7fced9091700) at pthread_create.c:463
        pd = 0x7fced9091700
        now = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {140526381242112, -2656864701086767078, 140526381239424, 0, 140526162564176, 140731283001648, 2647370783930902554, 2647322756688308250}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
#6  0x00007fcee516fa3f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
No locals.

Thread 3 (Thread 0x7fceda2a0700 (LWP 1599)):
#0  0x00007fcee5162cf9 in __GI___poll (fds=0x5592e39d35b0, nfds=1, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
        resultvar = 18446744073709551100
        sc_cancel_oldtype = 0
        sc_ret = <optimized out>
#1  0x00007fcee56a65c9 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#2  0x00007fcee56a66dc in g_main_context_iteration () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#3  0x00007fcee56a6721 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#4  0x00007fcee56ce175 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#5  0x00007fcee2d276db in start_thread (arg=0x7fceda2a0700) at pthread_create.c:463
        pd = 0x7fceda2a0700
        now = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {140526400177920, -2656864701086767078, 140526400175232, 0, 94089371912992, 140731283002560, 2647377651046737946, 2647322756688308250}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
#6  0x00007fcee516fa3f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
No locals.

Thread 2 (Thread 0x7fced9a9f700 (LWP 1600)):
#0  0x00007fcee5162cf9 in __GI___poll (fds=0x5592e39e38e0, nfds=2, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
        resultvar = 18446744073709551100
        sc_cancel_oldtype = 0
        sc_ret = <optimized out>
#1  0x00007fcee56a65c9 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#2  0x00007fcee56a6962 in g_main_loop_run () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#3  0x00007fcee63e9276 in ?? () from /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0
No symbol table info available.
#4  0x00007fcee56ce175 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#5  0x00007fcee2d276db in start_thread (arg=0x7fced9a9f700) at pthread_create.c:463
        pd = 0x7fced9a9f700
        now = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {140526391785216, -2656864701086767078, 140526391782528, 0, 94089371913216, 140731283002944, 2647369953928472602, 2647322756688308250}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
#6  0x00007fcee516fa3f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
No locals.

Thread 1 (Thread 0x7fcee76aba80 (LWP 1571)):
#0  0x00007fcee56abea1 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#1  0x00007fcee56ae819 in g_log_writer_default () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#2  0x00007fcee56aca8e in g_log_structured_array () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#3  0x00007fcee56ad4ce in g_log_structured_standard () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#4  0x00007fcee6719c41 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
No symbol table info available.
#5  0x00007fcee6726ac3 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
No symbol table info available.
#6  0x00007fcee5dce8fa in _XError () from /usr/lib/x86_64-linux-gnu/libX11.so.6
No symbol table info available.
#7  0x00007fcee5dcb82b in ?? () from /usr/lib/x86_64-linux-gnu/libX11.so.6
No symbol table info available.
#8  0x00007fcee5dcb8d5 in ?? () from /usr/lib/x86_64-linux-gnu/libX11.so.6
No symbol table info available.
#9  0x00007fcee5dcc830 in _XReply () from /usr/lib/x86_64-linux-gnu/libX11.so.6
No symbol table info available.
#10 0x00007fcee5db2435 in XGetWindowProperty () from /usr/lib/x86_64-linux-gnu/libX11.so.6
No symbol table info available.
#11 0x00007fcee6737064 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
No symbol table info available.
#12 0x00007fcee6737d55 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
No symbol table info available.
#13 0x00007fcee6720eef in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
No symbol table info available.
#14 0x00007fcee67211f6 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
No symbol table info available.
#15 0x00007fcee66ebd70 in gdk_display_get_event () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
No symbol table info available.
#16 0x00007fcee6720f82 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
No symbol table info available.
#17 0x00007fcee56a6417 in g_main_context_dispatch () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#18 0x00007fcee56a6650 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#19 0x00007fcee56a6962 in g_main_loop_run () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#20 0x00007fcee6bdea25 in gtk_main () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.
#21 0x00005592e1f9bf19 in main ()
No symbol table info available.
#0  0x00007fcee56abea1 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#1  0x00007fcee56ae819 in g_log_writer_default () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007fcee56aca8e in g_log_structured_array () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007fcee56ad4ce in g_log_structured_standard () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#4  0x00007fcee6719c41 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#5  0x00007fcee6726ac3 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#6  0x00007fcee5dce8fa in _XError () from /usr/lib/x86_64-linux-gnu/libX11.so.6
#7  0x00007fcee5dcb82b in ?? () from /usr/lib/x86_64-linux-gnu/libX11.so.6
#8  0x00007fcee5dcb8d5 in ?? () from /usr/lib/x86_64-linux-gnu/libX11.so.6
#9  0x00007fcee5dcc830 in _XReply () from /usr/lib/x86_64-linux-gnu/libX11.so.6
#10 0x00007fcee5db2435 in XGetWindowProperty () from /usr/lib/x86_64-linux-gnu/libX11.so.6
#11 0x00007fcee6737064 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#12 0x00007fcee6737d55 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#13 0x00007fcee6720eef in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#14 0x00007fcee67211f6 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#15 0x00007fcee66ebd70 in gdk_display_get_event () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#16 0x00007fcee6720f82 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#17 0x00007fcee56a6417 in g_main_context_dispatch () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#18 0x00007fcee56a6650 in ?? () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#19 0x00007fcee56a6962 in g_main_loop_run () from /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#20 0x00007fcee6bdea25 in gtk_main () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#21 0x00005592e1f9bf19 in main ()

**Comment by mprt1966** [comment]
[crash.tar.gz](https://github.com/linuxmint/cinnamon-settings-daemon/files/4975762/crash.tar.gz)

**Comment by leigh123linux** [comment]
Duplicate of #279

**Comment by mprt1966** [comment]
Thank you for your reply, but what do you mean by " you modified the open/close state" , I am not an expert of linux.
which file should I modify or install to fix the problem ?
Thank you.

**Comment by leigh123linux** [comment]
I closed it as your issue is a duplicate of #279
